.github/workflows/changelog-preview.yml

```diff
@@ -0,0 +1,109 @@
+name: Changelog Preview
+
+on:
+  # Allow this workflow to be called from other repositories
+  workflow_call:
+    inputs:
+      craft-version:
+        description: 'Version of Craft to use (tag or "latest")'
+        required: false
+        type: string
+        default: 'latest'
+
+  # Also run on PRs in this repository (for dogfooding)
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+permissions:
+  pull-requests: write
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install Craft
```

Comment:
You should be using the new common action from `install/action.yml` for installing Craft. If that's not viable currently try referencing the PR as its version (like `pull/669/head`) and then swap that once we merge this PR.